src/Migrations/Version20201208105207.php

```diff
@@ -0,0 +1,109 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\RefundPlugin\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * @internal
+ *
+ * This class is not covered by the backward compatibility promise and it will be removed after update Sylius to 1.9.
+ * It is a duplication of a logic from Sylius to provide proper adjustments handling.
+ */
+final class Version20201208105207 extends AbstractMigration
+{
+    /** @var array */
+    private $shippingName = [];
+
+    public function getDescription(): string
+    {
+        return 'Set details and shipment_id on shipping adjustments.';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->setDefaultAdjustmentData();
+
+        $adjustments = $this->getShipmentAdjustmentsWithData();
+
+        foreach ($adjustments as $adjustment) {
+            if (!isset($this->shippingName[$adjustment['shipment_code']])) {
+                $this->shippingName[$adjustment['shipment_code']] = $adjustment['label'];
+            }
+
+            $this->updateAdjustment(
+                (int) $adjustment['id'],
+                $adjustment['shipping_id'] ?? 'NULL',
+                $this->getParsedDetails(['taxRateCode' => $adjustment['tax_rate_code'], 'taxRateName' => $adjustment['tax_rate_name'], 'taxRateAmount' => ($adjustment['tax_rate_amount'] ? (float) $adjustment['tax_rate_amount'] : null), 'shippingMethodCode' => $adjustment['shipment_code'], 'shippingMethodName' => $this->getShippingMethodName($adjustment['shipment_code'])])
+            );
+        }
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->setDefaultAdjustmentData();
+    }
+
+    private function getShippingMethodName(?string $shippingMethodCode): ?string
+    {
+        if ($shippingMethodCode === null) {
+            return $shippingMethodCode;
+        }
+
+        return $this->shippingName[$shippingMethodCode];
+    }
+
+    private function getParsedDetails(array $details): string
+    {
+        /** @var array $parsedDetails */
+        $parsedDetails = [];
+
+        foreach ($details as $key => $value) {
+            if ($value !== null) {
+                $parsedDetails[$key] = $value;
+            }
+        }
+
+        /** @var string $encodedDetails */
+        $encodedDetails = json_encode($parsedDetails);
+
+        return $encodedDetails;
+    }
+
+    private function setDefaultAdjustmentData(): void
+    {
+        $this->connection->executeQuery("UPDATE sylius_adjustment SET shipment_id = null, details = '[]'");
+    }
+
+    private function updateAdjustment(int $adjustmentId, string $shipmentId, string $details): void
+    {
+        $this->connection->executeQuery("UPDATE sylius_adjustment SET shipment_id = $shipmentId, details = '" . $details . "' WHERE id = $adjustmentId");
```

Comment:
Why did you decide for query execution instead of addSql? I would have to dig into it, but IIRC `addSql` is a good practice with the doctrine migrations

Reply:
This is a copy-paste migration from [this PR](https://github.com/Sylius/Sylius/pull/12116)

Reply:
This may collapse during dry-run execution (ref. https://stackoverflow.com/a/18580775). Can we adjust it to `addSql` in both places?